libs/libcommon/include/common/sse2_memcpy.h

```diff
@@ -50,21 +50,21 @@ ALWAYS_INLINE static inline void * sse2_inline_memcpy(void * __restrict dst_, co
             }
             /// No bytes remaining.
         }
-        else if (unlikely(size <= 4))
+        else if (unlikely(size < 4)) // sse2_inline_memcpy(_,_, 4 ) should use 4 bytes register directly
```

Comment:
Because the behavior of compiler is not very well.
For example: [godbolt.org](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM9KuADJ4DJgAcj4ARpjEEhoAnKQADqgKhE4MHt6%2B/tKp6Y4CIWGRLDFxXIm2mPZFDEIETMQE2T5%2BATV1mY3NBCUR0bHxSQpNLW25nWN9A2UVIwCUtqhexMjsHOYAzKHI3lgA1CbbbpgsLKEExKEAdAgn2CYaAII7eweYx6fIY%2BhYVEezzeZl2DH2XiOJzcvwI%2BEEQNewJ2h1ReCohzAYABoUw6AgtDwAGtagBPRbI0Go6monFhQ6Ekm0UkQVSLakQAD6nKiXjojgYnMwqmSmAcECxWNZy0OXEWFKRVNRrnRlO21PRmOxmBoYXxXgYjLJCpB6pptJ1uMOBqNzOlhy5PL5tAFQpFYoIEsl0tIhw08rV1JVgKRrxmjmQh1ChPpADdUHh0IcAFSHBRKMyc6O4zlsFjIZIs%2BOJlOHbnETBjG4OQ7oMac31oBhjQ7FpOp8uV654GsKdYNtN4ABemE5BEHI5NJgA7FZXjTkAhmqXO1We%2BO6%2BOTgARQ4Vq6xZIVgic0RjaGL5cdzkVtcOR4QTecinbOcvBcCFuX4grm9d6vjn2kY7numAHsQR6YCeZ4EBen7jt%2Bv63t297bNgEBAc%2BJxvsCNJtqWx7fLum7Yciiq7BidJ4hAACybjKOyM5uExtaWnqtH0Zy2DhNuJo0uGPackw6axJ6VBiEoL44aCwZqlRhyOFQBgKAgp6oCwyR0LETr8qEubnAWpJljpLp6XmhlZoaVoAPTWcZvK6YK5mFpZMZfMAayGAQmCVgpS4Iepmn0D%2BADudC0IcDCoOO5yEIcoi0BFRAMngUTIGAHAKIczmknJbFfHRyhcTxEAMOyinKapaAaVpxAma6OWPi21gTl8AC0Dplb6QHHJYrWHB1pUykNUl5bqBX0UNCnopVak1cF9VmQZhZNQQ3XrL6I2kaG74ahiBLEmSGHDl80IgdsZgBvO1Izm%2B5p7Q6NqHXa6Qjt8zHbLucp8fdxyzrhv00pqB1Miyr2nZ9IHfQDgOordMOwzStl2e%2B6RGPQhxRKS3m3AjiOosmm5ESmQHbbtiMztueO/cjhzhKgmPY75FYsEwoShMAuPXfdlPU8qtBKFG%2B1PaDx1vdChySPKhy2WmGauTmjUNgOkiHOyKmrLQSZeILqtY95WUVsAeBjLEtZ4BWDjMnzf13fjtNuAgBpEllqAYmYty3Oq%2BuVlz5Ow4VECXWT%2BOHIVxXbkHo3c%2BavMxzStSC8DItHeD72HAAHNLssiZm2ZhPp%2BYrcrvoZ2raYIJr2uC2XPuG5gxumz%2B%2BCWwQ1vxzd/0d/dDtOwwLuHG7kue9OjMG37oeB1LIf4%2BH3GR9Pr423H/sJwLXzJ1Zotp2dn2ygAbFdq%2Bd3biO987rsYhnntcPvY%2B%2BzbNKB1nM%2BI3PJUv0v3d/VT3%2BJ5gy8u7Hx7rLR2F9B4Yi4NOT2F174KAnrPSat9o7APNO/SOyDX6x2nL/f2K8aTHjWAwUCsEv5vHIgaAEYd6JqkoTqahRV57IhwWRF4AlIz5y%2BPhVMucFYFxypyEgeAIDcOMshACtZ6yNngq2BM7YxH/nXGmfs3UTpjlalOIBC4lw/mvOIpRRMQL7kEIeY8p5hKkJhDopCijULoSfCgj8zYELWL0bYwC6xibGO8hBMxME4LOPiq4hRd5SHoUwighGojCIgRImQtUmoqL4kKoxaczE0msXGskzi88fqonYUJESLRqASUwJEmSDB8AhlNPJCqwkqqBVqotJyy0jLcgcqZFpRdSR8K%2BLLdpzoGqtN6YcDyxAvI%2BSygQfy8VGnBUOGFRKkVoqHFighMQSUGaEjShlLKOUxpWnQVNOpKk5pBW0h0oZ3TVq9QsP1QaXVlHAT6mnB5w0yrlO2PJQOZVppKXqWcpplylrXM3OtZAm0PlkwRsDW0YMTrvXOpdPJtsbabzhWLCGH0vpH0BvDb%2BQN9oYp3pDPe0MCUn0fjTWWqMOYYx9gg0OBNDF72TKTMhod8H21lvTOBoFWbsyMIy7BuDAb/yFo9LeqcEW713FLFFNJ8WoKRqAvuA8h4e1uHrJm8CqWokDsHDliDGElUNafEVNtxXouevC8WpwQJZwVZSilqJz790vocAArJ7WuOrhWAyno441EcICL3NYqlhf914SogCnF6Mr7VksPk6uGWjuV2TAe6iBhwEg3zvnXf1v1n5Brfjkj%2BJaLVRqUIA8N1KM1qo9VAmBZg4GFvuoHTBRrS0mowYfLBEbRWd1wQjQhxBiHHmhRQyp9DCq0OnRidBzDh07VEQbE8vDOGFwshnQRNwRFyJsaEyRJ5pGBNEauFCHjkBYSRGm1Eo7iEbqsvw4ZQjVoDkwqXSJkbyEvFXV2TkT63JbpchnfdJY3FHqfKels56/xHoicw81D65aYDzs%2B0cjVoNPIHJ/HCP6ODLFoJwT1vA/AcC0KQVAnBmKWBagoVY6xTqgh4KQAgmhCPLCJCASQkhbjTm2Bnfe05PUJE9RnLgGcJOev0JwSQZGONUc4LwBQIANBsY48sOAsAYCIBQHM2I5BKDVXOXEfYhhgBcG2BodTNAXSxFUxAKIimoihGaKSTgrHqpsEEAAeUNB5ijvAsCs3RpsSjLcPR4FjJWRTwoxReG8opg8xGgv6FSuM4gpIPBYEU92FgnnCN8AMMABQAA1PAmAQq%2BdFOR1j/BBAiDEOwKQMhBCKBUOoNLuguD6Asygaw1h0tRFU7AZgbAQBUBYGtGLcRrjO3lFR5I9RVMcDar59UbU/g7lMHRywUCBrbh4tgCwcgADiKnahRecBAVwkw/C9eCLieYwxesFAyAIe7eh3v1DmEMSoXRrsNHGK0Tw7Q9B2CB70Fof3yivdsCDr7vWZgw%2Be/9iQywGNrA2Hoa4PlCtEZIwptL1GOCqCE21feqtzNGFlNsW4GgGcOlo1YSwvpcCEBIL1bYvXDgeHmmbHYcpeDsaC4t7jnr1Opfk6QArGdvUJHl9OSQGczCSc9WYATZhSDkco6TlTamNNi9INpvTqwCDJES0Z27%2BAiA3HQHoBrwgEotds7IDrahFM9dICFcZyQCeyY4KRnXinSe%2BcS5b8cQ9yf70p9TgwtOrMM6ZxAfnpnufC6N1oRbCBMBMCwHECAhOOAy4K1IBnUgzCeqgRJyQ04zD7wztOEPJPlO2EN6L7PXGQCeu9Q3qzAQNCSEk1wdX0hUvbGJ3rtvnfOOB7MFP3g%2Bus9z9m%2BkZwkggA%3D)

If set size to 8, original behavior like `test_sse2_inline_memcpy_8_ori` still need two memory move by register (4B).

Reply:
awesome